remote/new.go

```diff
@@ -154,7 +156,17 @@ func processPreviousImageOption(ri *Image, prevImageRepoName string, platform im
 		return errors.Wrapf(err, "getting layers for previous image with repo name %q", prevImageRepoName)
 	}
 
+	configFile, err := prevImage.ConfigFile()
+	if err != nil {
+		return err
+	}
+
 	ri.prevLayers = prevLayers
+	prevHistory := configFile.History
+	if len(prevLayers) != len(prevHistory) {
+		prevHistory = make([]v1.History, len(prevLayers))
```

Comment:
Alternatively we could see if removing history for "empty layers" will cause the lengths to be equal